Log is filled with errors when using a Favorite with Ampersand in Sonos Binding

Hi,

when you specify an Sonos-Favorite you want to play which contains an Ampersand (&), you get lots of erros in the logs and console:
`sendCommand(Sonos_Favorite,"Entspannte Rock-Klassiker & Oldies")`
leads to
`[ERROR] [inding.sonos.internal.SonosXMLParser] - Could not parse MetaData from string '<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="100c206ccatalog%2fstations%......%2f%23chunk-....." parentID="10012064catalog%2fstations%2frefinements%2fgenres%2fa3d0cc3b-368f-465e-b995-fasdafs%2f%23prime_stations" restricted="true"><dc:title>Entspannte Rock-Klassiker & Oldies</dc:title><upnp:class>object.item.audioItem.audioBroadcast</upnp:class><desc id="cdudn" nameSpace="urn:schemas-rinconnetworks-com:metadata-1-0/">SA_RINCON51363_X_#Svc533463-0-Token</desc></item></DIDL-Lite>`
and in the console to `[Fatal Error] :1:481: The entity name must immediately follow the '&' in the entity reference` which is repeated every few seconds.

The Favorite is getting played though. I have tried it with several escapings but none of them worked. The Favorite is not found and nothing is getting played.

```
Favorite 'Entspannte Rock-Klassiker &amp; Oldies' not found
Favorite 'Entspannte Rock-Klassiker &amp;amp; Oldies' not found
Favorite 'Entspannte Rock-Klassiker %26 Oldies' not found
Favorite 'Entspannte Rock-Klassiker &#38; Oldies' not found
```

I can still reproduce this issue. 
If I use `sendCommand(Sonos_Favorite,"Entspannte Rock-Klassiker & Oldies") ` inside a rule I see in the openhab.log the following output:

`2017-08-20 18:37:16.144 [ERROR] [inding.sonos.internal.SonosXMLParser] - Could not parse MetaData from string '<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="135c106ccatalog%2fstations%2fA3F22YHFVQ9TGY%2f%23chunk-sQfKS2mBSZusTMSUcgiQ" parentID="10082064catalog%2fstations%2frefinements%2fgenres%2fa3d0cc3b-368f-465e-b995-2212e61fe93b%2f%23prime_stations" restricted="true"><dc:title>Entspannte Rock-Klassiker & Oldies</dc:title><upnp:class>object.item.audioItem.audioBroadcast</upnp:class><desc id="cdudn" nameSpace="urn:schemas-rinconnetworks-com:metadata-1-0/">SA_RINCON74363_X_#Svc51263-0-Token</desc></item></DIDL-Lite>`

and in the Openhab Console log the following every few seconds:

`[Fatal Error] :1:481: Auf "&" in der Entityreferenz muss umgehend der Entityname folgen.
`

Entspannte Rock-Klassiker & Oldies is an Amazon Prime Radio station. 

It seems to be sth with the station or the thing that this is a favorite inside Sonos. If I create a playlist and name it "A&B" and start it with `sendCommand(Sonos_Playlist,"A & B")` it gets played without an error

--- a/extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/internal/SonosXMLParser.java
+++ b/extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/internal/SonosXMLParser.java
@@ -20,6 +20,7 @@ import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import org.apache.commons.lang.StringEscapeUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.xml.sax.Attributes;
@@ -948,6 +949,8 @@ public class SonosXMLParser {
             desc = entry.getResourceMetaData().getDesc();
             upnpClass = entry.getResourceMetaData().getUpnpClass();
         }
+
+        title = StringEscapeUtils.escapeXml(title);
 
         String metadata = METADATA_FORMAT.format(new Object[] { id, parentId, title, upnpClass, desc });
 

